src/components/HistoryTable/utils.ts

```diff
@@ -12,10 +13,18 @@ import { mainlineCommits, CommitRowType, rowType } from "./types";
 // - A set of commits that are rolled up into a single commit
 // The processed commits are used to calculate the height of each row in the table.
 // They are ordered by the order of the commits in the table.
-export const processCommits = (
-  newCommits: mainlineCommits["versions"],
-  existingCommits: CommitRowType[]
-) => {
+export const processCommits = ({
+  newCommits,
+  existingCommits,
+  selectedCommitOrder,
+  selectedCommitRow,
+}: {
+  newCommits: mainlineCommits["versions"];
+  existingCommits: CommitRowType[];
+  selectedCommitOrder: number | null;
+  selectedCommitRow: number | null;
+}) => {
+  let selectedCommitRowIndex: number | null = selectedCommitRow;
   const processedCommits: CommitRowType[] = [...existingCommits];
   for (let i = 0; i < newCommits.length; i++) {
```

Comment:
We would have had to run this function in order to generate the rows in the first place. So this code had to have been ran on the initial rows.

Reply:
That makes sense that his function is ran after new commits are fetched. The rowIndex won't be calculated if new commits don't need to be loaded. This is why the row isn't highlighted when selecting 2 buildvariant labels that belong to the same response page. We should separate the rowIndex calculation from the commit processing.

Reply:
How would the initial rows be rendered if it does not run this?  The  `processedCommits` are stored in a context which wraps this page. If you navigate away and return the context will be a fresh instance. It will have to rerun this function in order to process the data into rows regardless of how they were fetched. 
You can see this behavior by adding a `console.log` within this function. It will step through each version no matter which way it was fetched. (A cached copy from the apollo cache from a previous page or a new copy fetched over the network). For what ever reason the strict `===` comparison is not matching the `selectedCommitOrder` with the `version.order` when we compare it against the cached copy. Yet everything else is working.

Reply:
Discussed offline opened https://jira.mongodb.org/browse/EVG-16642